dgl_sparse/src/matrix_ops.cc

```diff
@@ -58,12 +55,112 @@ std::tuple<std::shared_ptr<COO>, torch::Tensor, torch::Tensor> COOIntersection(
   return {ret_coo, lhs_indices, rhs_indices};
 }
 
+torch::Tensor RevertIndices(const torch::Tensor& indices) {
+  auto rev_tensor = torch::empty_like(indices);
+  rev_tensor.index_put_(
+      {indices}, torch::arange(0, indices.numel(), rev_tensor.options()));
+  return rev_tensor;
+}
+
+std::tuple<torch::Tensor, torch::Tensor> CompactIndices(
```

Comment:
This function is complex and used multiple times. Please add comments to explain its functionality.